packages/ui/__tests__/components/organisms/CheckboxScrollList.test.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2023 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import {View} from 'react-native';
+
+import {shallow} from 'enzyme';
+import {CheckboxScrollList, ScrollList, Checkbox} from '@axelor/aos-mobile-ui';
+
+describe('CheckboxScrollList Component', () => {
+  const mockData = [
+    {id: 1, name: 'Item 1'},
+    {id: 2, name: 'Item 2'},
+  ];
+  const onCheckedChange = jest.fn();
+  const renderItem = jest.fn(item => <View>{item.name}</View>);
+
+  const props = {
+    data: mockData,
+    onCheckedChange,
+    renderItem,
+  };
+
+  it('renders without crashing', () => {
+    const wrapper = shallow(<CheckboxScrollList {...props} />);
+    expect(wrapper.exists()).toBe(true);
+  });
+
+  it('toggles all items when main checkbox is clicked', () => {
+    const wrapper = shallow(<CheckboxScrollList {...props} />);
+    wrapper.find(Checkbox).at(0).simulate('change', true);
+    expect(onCheckedChange).toHaveBeenCalledWith(mockData);
+    wrapper.find(Checkbox).at(0).simulate('change', false);
+    expect(onCheckedChange).toHaveBeenCalledWith([]);
+  });
+
+  it('toggles individual items', () => {
+    const wrapper = shallow(<CheckboxScrollList {...props} />);
+    wrapper
+      .find(ScrollList)
+      .renderProp('renderItem')({item: mockData[0], index: 0})
+      .find(Checkbox)
+      .simulate('change', true);
+    expect(onCheckedChange).toHaveBeenCalledWith([mockData[0]]);
+  });
```

Comment:
I think you can add two more actions in this test : 
* simulate check on another item to check that item is correctly added and do not replace old one
* simulate check on a selected item to check that item is correctly removed.